agent/__init__.py

```diff
@@ -465,155 +467,156 @@ def setup_endpoints(dev, debug):
 
 
 def fetch_credentials(debug, dev):
-
-    setup_endpoints(dev, debug)
-    print('Fetching credentials...')
-    can_read_cert()
-
-    credentials_req = requests.get(
-        '{}/v0.2/creds'.format(MTLS_ENDPOINT),
-        cert=(CLIENT_CERT_PATH, CLIENT_KEY_PATH),
-        headers={
-            'SSL-CLIENT-SUBJECT-DN': 'CN=' + get_device_id(),
-            'SSL-CLIENT-VERIFY': 'SUCCESS'
-        } if dev else {}
-    )
-    if not credentials_req.ok:
-        print('Fetching failed.')
+    with Locker('credentials'):
+        setup_endpoints(dev, debug)
+        print('Fetching credentials...')
+        can_read_cert()
+
+        credentials_req = requests.get(
+            '{}/v0.2/creds'.format(MTLS_ENDPOINT),
+            cert=(CLIENT_CERT_PATH, CLIENT_KEY_PATH),
+            headers={
+                'SSL-CLIENT-SUBJECT-DN': 'CN=' + get_device_id(),
+                'SSL-CLIENT-VERIFY': 'SUCCESS'
+            } if dev else {}
+        )
+        if not credentials_req.ok:
+            print('Fetching failed.')
+            if debug:
+                print("[RECEIVED] Fetch credentials: code {}".format(credentials_req.status_code))
+                print("[RECEIVED] Fetch credentials: {}".format(credentials_req.content))
+            return
+        credentials = credentials_req.json()
+
+        print('Credentials retreived.')
         if debug:
-            print("[RECEIVED] Fetch credentials: code {}".format(credentials_req.status_code))
-            print("[RECEIVED] Fetch credentials: {}".format(credentials_req.content))
-        return
-    credentials = credentials_req.json()
+            print('Credentials: {}'.format(credentials))
 
-    print('Credentials retreived.')
-    if debug:
-        print('Credentials: {}'.format(credentials))
+        if not os.path.exists(CREDENTIALS_PATH):
+            os.mkdir(CREDENTIALS_PATH, 0o700)
 
-    if not os.path.exists(CREDENTIALS_PATH):
-        os.mkdir(CREDENTIALS_PATH, 0o700)
+        if not os.path.isdir(CREDENTIALS_PATH):
+            print("there is file named as our credentials dir({}), that's strange...".format(CREDENTIALS_PATH))
+            exit(1)
 
-    if not os.path.isdir(CREDENTIALS_PATH):
-        print("there is file named as our credentials dir({}), that's strange...".format(CREDENTIALS_PATH))
-        exit(1)
+        for f in os.listdir(os.path.join(CREDENTIALS_PATH)):
```

Comment:
Let's try to avoid single character variable names.

Reply:
@vpetersson I didn't write this. It was written in #107

Reply:
Sure, but that doesn't mean we can't fix it now. We should always fix issues when we run across them, regardless who the original author said code is.